docs/tasks/administer-cluster/kubeadm-upgrade-ha.md

```diff
@@ -0,0 +1,176 @@
+---
+reviewers:
+- jamiehannaford 
+- luxas
+- timothysc 
+- jbeda
+title: Upgrading kubeadm HA clusters from 1.9.x to 1.9.y
+---
+
+{% capture overview %}
+
+This guide is for upgrading `kubeadm` HA clusters from version 1.9.x to 1.9.y where `y > x`. The term "`kubeadm` HA clusters" refers to clusters of more than one master node created with `kubeadm`. To set up an HA cluster for Kubernetes version 1.9.x `kubeadm` requires additional manual steps. See [Creating HA clusters with kubeadm](/docs/setup/independent/high-availability/) for instructions on how to do this. The upgrade procedure described here targets clusters created following those very instructions. See [Upgrading/downgrading kubeadm clusters between v1.8 to v1.9](/docs/tasks/administer-cluster/kubeadm-upgrade-1-9/) for more instructions on how to create an HA cluster with `kubeadm`.
+
+{% endcapture %}
+
+{% capture prerequisites %}
+
+Before proceeding:
+
+- You need to have a functional `kubeadm` HA cluster running version 1.9.0 or higher in order to use the process described here.
+- Make sure you read the [release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.9.md) carefully.
+- Note that `kubeadm upgrade` will not touch any of your workloads, only Kubernetes-internal components. As a best-practice you should back up anything important to you. For example, any application-level state, such as a database and application might depend on (like MySQL or MongoDB) should be backed up beforehand.
+- Read [Upgrading/downgrading kubeadm clusters between v1.8 to v1.9](/docs/tasks/administer-cluster/kubeadm-upgrade-1-9/) to learn about the relevant prerequisites.
+
+{% endcapture %}
+
+{% capture steps %}
+
+## Preparation
+
+Some preparation is needed prior to starting the upgrade. First download the version of `kubeadm` that matches the version of Kubernetes that you are upgrading to:
+
+```shell
+# Use the latest stable release or manually specify a
+# released Kubernetes version
+export VERSION=$(curl -sSL https://dl.k8s.io/release/stable.txt) 
+export ARCH=amd64 # or: arm, arm64, ppc64le, s390x
+curl -sSL https://dl.k8s.io/release/${VERSION}/bin/linux/${ARCH}/kubeadm > /tmp/kubeadm
+chmod a+rx /tmp/kubeadm
+```
+
+Copy this file to `/tmp` on your primary master if necessary. Run this command for checking prerequisites and determining the versions you will receive:
+
+```shell
+/tmp/kubeadm upgrade plan
+```
+
+If the prerequisites are met you'll get a summary of the software versions kubeadm will upgrade to, like this:
+
+    Upgrade to the latest stable version:
+
+    COMPONENT            CURRENT   AVAILABLE
+    API Server           v1.9.0    v1.9.2
+    Controller Manager   v1.9.0    v1.9.2
+    Scheduler            v1.9.0    v1.9.2
+    Kube Proxy           v1.9.0    v1.9.2
+    Kube DNS             1.14.5    1.14.7
+    Etcd                 3.2.7     3.1.11
```

Comment:
Is an etcd downgrade expected and intentional here?

Reply:
This is confusing, but seems unavoidable here: we don't upgrade etcd, since it is hosted externally. However `kubeadm upgrade plan` checks etcd versions anyway and prints them in the summary. There's a "caution" box under this example explaining that upgrading etcd is beyond scope. To me this seems sufficient.